Adding spec for Date/TimePicker design updates (#114)
microsoft/microsoft-ui-xaml-specs: active/DateTimePicker/DateTimePicker-Visual-Updates-Spec.md

[NavigationView](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/navigationview).

Recently, [ListView was given design updates](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/anawish/ListViewGridViewDesignUpdates/active/ListViewGridView/listviewgridview-designupdates.md) to round the corners of its items. DatePicker and TimePicker should follow in these footsteps, and make sure that they adopt any of the ListView design elements that are relevant. This includes general rounding of selection visuals, as well as rounded grey backplates for items that get hovered. 

This spec will detail a number of changes to update the designs of DatePicker and TimePicker so that they are more 
visually aligned with modern WinUI controls. At a high level, some key parts of these updated styles are: a rounded selection rectangle; rounded backplates on items, buttons, and scroll buttons on  hover; and new animations that are aligned with other controls. 

> There is only one major API change that will come with these styling updates, which is adding a new class called `ColorFilterOverlayControl`. This class provides the logic behind displaying the selection rectangle that spans across all three rows in DatePicker and TimePicker, and giving it the correct background color. 

Reviewer: ```suggestion
There is only one major API change that will come with these styling updates, which is adding a new class called `ColorFilterOverlayControl`. This class provides the logic behind displaying the 'inverted' selection rectangle that spans across all three columns in DatePicker and TimePicker, and giving it the correct foreground color to have appropriate contrast over the accent colored background.
```

![Shows the old TimePicker visuals alongside the new TimePicker visuals](images/timepicker-evolution.PNG)

**Rounded backplates and focus visuals**

As you can see in the mockup below, all buttons and items (individual cells) within DatePicker and TimePicker will receive a rounded grey backplate when hovered or pressed. In addition to this, columns within DatePicker and TimePicker will receive rounded focus rectangles (as shown below).

![Shows a DatePicker with scroll and check buttons being hovered, as well as the last column being focused upon](images/DPTP-other-states.PNG)

# API Notes
<!-- Option 1: Give a one or two line description of each API (type
and member), or at least the ones that aren't obvious
from their name.  These descriptions are what show up
in IntelliSense. For properties, specify the default value of the property if it
isn't the type's default (for example an int-typed property that doesn't default to zero.) -->

<!-- Option 2: Put these descriptions in the below API Details section,
with a "///" comment above the member or type. -->

## New APIs

`public Class ColorFilterOverlayControl`

> Represents a class that takes a rectangle and transforms into the coordinate space of a TargetElement, and then runs a ColorMatrixEffect on that region to select the alpha channel and replace with a ReplacementColor.

Reviewer: I don't know how to phrase this better but I'll give some other words in the hopes that someone can help edit this into documentation-appropriate terms:

This class assists in doing a specialized color filter applied to another element (the TargetElement). Using whatever rectangle of the ColorFilterOverlayControl overlaps the TargetElement, the ColorFilterOverlayControl renders that region of the TargetElement to an offscreen image and passes that image through a ColorMatrixEffect which takes the Alpha channel for each pixel of the TargetElement and replaces the RGB channels with the RGB of the ReplacementColor.

Author: @jevansaks How about this?

Represents a class that assists in applying a specialized color filter onto a TargetElement. The region of the TargetElement that has overlap with the ColorFilterOverlayControl is rendered to an offscreen image that's passed through a ColorMatrixEffect, which takes the Alpha channel of each pixel in the region. RGB channels are then replaced with the RGB of a ReplacementColor.

Jevan/@MikeHillberg Is this too long for a general description that'll be used in Intellisense? Perhaps only the first sentence should be use for Intellisense and the rest should be on the API reference page to provide more context.

Reviewer: That description sounds good to me. For intellisense usually the first sentence of the description is used in the tooltip, and I think the first sentence you have is fine.

`public UIElement TargetElement { get; set; }`

> This is a property of `ColorFilterOverlayControl`. Gets or sets the UIElement whose coordinate space should be replaced with a color. 

Reviewer: ```suggestion
This is a property of `ColorFilterOverlayControl`. Gets or sets the UIElement to apply the color effect to. 
```

`public Color ReplacementColor { get; set; }`

> This is a property of `ColorFilterOverlayControl`. Gets or sets the Color that should fill the target element. 

Reviewer: ```suggestion
This is a property of `ColorFilterOverlayControl`. Gets or sets the Color to use instead of the RGB channels of the TargetElement.
```

> This is a property of `ColorFilterOverlayControl`. Gets or sets the Color that should fill the target element. 

Reviewer 2: This line is what should become the Intellisense text, so we don't need "This is a property of ...".

## New ThemeResources

```xml
<StaticResource x:Key="DatePickerFlyoutPresenterHighlightForegroundColor" ResourceKey="TextOnAccentAAFillColorPrimary" />
```

# API Details
<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->

```csharp

> namespace MU_XCP_NAMESPACE

Reviewer: ```suggestion
namespace Microsoft.UI.Xaml.Controls
```

{

> [MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
> [MUX_PROPERTY_CHANGED_CALLBACK_METHODNAME("OnPropertyChanged")]

Reviewer: ```suggestion
```

API review doesn't need to know about this implementation detail. :)

Reviewer 2: Agreed, but I've been leaving them in; I don't think it causes harm, and it's nice to be able to copy/paste.

Author: Going to leave as is, let me know if there are objections.

> unsealed runtimeclass ColorFilterOverlayControl : Windows.UI.Xaml.Controls.Grid

Reviewer 2: Can we get an example usage?

Author: Added in upcoming commit, let me know if it's not detailed enough.
